fastcrypto/src/bulletproofs.rs

```diff
@@ -51,99 +56,22 @@ impl PedersenCommitment {
         let generators = PedersenGens::default();
         let value = Scalar::from_bits(value);
         let blinding = Scalar::from_bits(blinding_factor);
-        let point = generators.commit(value, blinding);
-
-        PedersenCommitment {
-            point,
-            bytes: OnceCell::new(),
-        }
-    }
-}
-
-impl ops::Add<PedersenCommitment> for PedersenCommitment {
-    type Output = PedersenCommitment;
-
-    fn add(self, rhs: PedersenCommitment) -> PedersenCommitment {
-        PedersenCommitment {
-            point: self.point + rhs.point,
-            bytes: OnceCell::new(),
-        }
+        generators.commit(value, blinding).into()
     }
 }
 
-impl ops::Sub<PedersenCommitment> for PedersenCommitment {
-    type Output = PedersenCommitment;
-
-    fn sub(self, rhs: PedersenCommitment) -> PedersenCommitment {
-        PedersenCommitment {
-            point: self.point - rhs.point,
-            bytes: OnceCell::new(),
-        }
-    }
-}
-
-impl AsRef<[u8]> for PedersenCommitment {
-    fn as_ref(&self) -> &[u8] {
-        self.bytes.get_or_init(|| self.point.compress().to_bytes())
-    }
-}
-
-impl ToFromBytes for PedersenCommitment {
-    fn from_bytes(bytes: &[u8]) -> Result<Self, FastCryptoError> {
-        if bytes.len() != PEDERSEN_COMMITMENT_LENGTH {
-            return Err(FastCryptoError::InputLengthWrong(
-                PEDERSEN_COMMITMENT_LENGTH,
-            ));
-        }
+impl ToFromByteArray<PEDERSEN_COMMITMENT_LENGTH> for PedersenCommitment {
+    fn from_byte_array(bytes: &[u8; PEDERSEN_COMMITMENT_LENGTH]) -> Result<Self, FastCryptoError> {
         let point = CompressedRistretto::from_slice(bytes);
-        let decompressed_point = point.decompress().ok_or(FastCryptoError::InvalidInput)?;
-
-        Ok(PedersenCommitment {
-            point: decompressed_point,
-            bytes: OnceCell::new(),
-        })
-    }
-}
-
-impl Serialize for PedersenCommitment {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        let bytes = self.as_ref();
-        serializer.serialize_bytes(bytes)
-    }
-}
-
-impl<'de> Deserialize<'de> for PedersenCommitment {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: de::Deserializer<'de>,
-    {
-        let bytes = Vec::deserialize(deserializer)?;
-        PedersenCommitment::from_bytes(&bytes[..]).map_err(|e| de::Error::custom(e.to_string()))
+        point.decompress().ok_or(InvalidInput).map(Self::from)
     }
-}
 
-impl PartialEq for PedersenCommitment {
-    fn eq(&self, other: &Self) -> bool {
-        self.point == other.point
+    fn to_byte_array(&self) -> [u8; PEDERSEN_COMMITMENT_LENGTH] {
+        self.point.compress().to_bytes()
     }
 }
 
-impl Eq for PedersenCommitment {}
-
-impl PartialOrd for PedersenCommitment {
```

Comment:
Are these (`PartialOrd` and `Ord`) really needed?

Reply:
is this for sui later? we dont have to add now

Reply:
Lets add them later if they are needed.